src/main/scala/com/johnsnowlabs/nlp/pretrained/ResourceDownloader.scala

```diff
@@ -589,9 +590,14 @@ object ResourceDownloader {
 
     val (bucketName, keyPrefix) = ResourceHelper.parseS3URI(path)
 
-    val tmpDirectory = SparkFiles.getRootDirectory()
+    val (accessKey, secretKey, sessionToken) = ConfigHelper.getHadoopS3Config
+    val privateS3Defined = accessKey != null && secretKey != null && sessionToken != null
```

Comment:
To be 100% sure all private arguments are defined. I will also check the region has data i.e. check `ConfigHelper.awsExternalRegion` is set

Reply:
I have included a check for region now. Also added a log line so that when config is not set correctly the user will be notified:

https://github.com/JohnSnowLabs/spark-nlp/pull/13432/files#diff-053c39e8ab751a99f5c1bb1da5c3b4bfde8954313ea782311a8bdecb4ca44114R593-R606